### PR TITLE
Fixes #8592. Add camel-quarkus-jsoup extension with native support   

### LINKED ISSUE #8592: Camel 4.21 - camel-jsoup extension
### Describe the feature here

https://issues.apache.org/jira/browse/CAMEL-23332

For CSB it was just a facade starter
https://github.com/apache/camel-spring-boot/commit/d2b51918daf4892d880984b2fd70213c4c3140ce

All there is in camel-jsoup is a set of simple functions. For example camel-base64 also have a set of simple functions so take a look there

### PR DESCRIPTION
**Pull request by jomin7**
Fixes #8592. Add camel-quarkus-jsoup extension with native support                                                                                        
                                                                                                                                                            
  Introduces Quarkus extension for Apache Camel jsoup component enabling
  HTML parsing, sanitization, and text extraction with GraalVM native support.                                                                              
                                                                                                                                                            
  - Minimal processor implementation (feature registration only)
  - Comprehensive tests: 10 test cases covering JVM, native, and edge cases                                                                                 
  - Simple language functions: htmlClean(), htmlDecode(), htmlParse()                                                                                       
  - YAML DSL support with route definitions                                                                                                                 
  - No reflection configuration required (auto-discovered by core)                                                                                          
  - Added to BOM, catalog, and test categories                                                                                                              
  - Generated documentation with extension metadata                                                                                                         
                                                                                                                                                            
  The extension provides XSS-safe HTML sanitization, text extraction,                                                                                       
  and DOM manipulation capabilities for Camel routes.

### CODE REVIEW

**Comment by jamesnetherton** [comment]
@jomin7 you'll need to rebase. I had to force push `camel-main` to get it back into a buildable state.

**Comment by jomin7** [comment]
Thanks for the heads up @jamesnetherton I'll rebase my branch onto the updated camel-main and force-push the changes shortly.

**Comment by jomin7** [comment]
I noticed the CI build failed on the latest push. I'm looking into the logs now and will push a fix shortly.

File: integration-tests/jsoup/src/test/java/org/apache/camel/quarkus/component/jsoup/it/JsoupTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+@QuarkusTest
+class JsoupTest {
+
+    @Test
+    public void testHtmlClean() {
```
Comment (jamesnetherton) [inline integration-tests/jsoup/src/test/java/org/apache/camel/quarkus/component/jsoup/it/JsoupTest.java]: Nitpick - remove public modifier from test methods.

Comment (jomin7) [reply integration-tests/jsoup/src/test/java/org/apache/camel/quarkus/component/jsoup/it/JsoupTest.java]: Thank you for the feedback. I'll go ahead and remove the public modifiers from the test methods.

**Comment by jomin7** [comment]
It appears an issue on the camel main branch, likely a recent change, is causing our current build failure. The integration-tests/grok module is failing to compile due to a missing io.krakens.grok.api.exception.GrokException class.

**Comment by apupier** [comment]
> It appears an issue on the camel main branch, likely a recent change, is causing our current build failure. The integration-tests/grok module is failing to compile due to a missing io.krakens.grok.api.exception.GrokException class.

there was a library change for grok, it should be "only" a matter of changing the package name from io.krakens.grok to io.whatap.grok , maybe adapt some tests https://github.com/apache/camel/pull/23031

**Comment by jamesnetherton** [comment]
@jomin7 I pushed some fixes. Please rebase again.

**Comment by jomin7** [comment]
> @jomin7 I pushed some fixes. Please rebase again.

@jamesnetherton I have rebased the changes on the updated camel-main branch.

**Review by jamesnetherton — APPROVED** [review]
